_ingest-pipelines/processors/foreach.md

```diff
@@ -0,0 +1,185 @@
+---
+layout: default
+title: Foreach
+parent: Ingest processors
+nav_order: 110
+---
+
+# Foreach processor
+
+The `foreach` processor is used to iterate over a list of values in an input document and perform some operation on each value. This can be useful for tasks like extracting information from a nested JSON structure or applying transformations to a collection of fields.
+
+The following is the syntax for the `foreach` processor:
+
+```json
+{
+  "foreach": {
+    "field": "<field_name>",
+    "processor": {
+      "<processor_type>": {
+        "<processor_config>": "<processor_value>"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `foreach` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The array field to iterate over.
+`processor` | Required | The processor to execute against each field.
+`ignore_missing` | Optional | If `true` and the specified field does not exist or is 
+null, the processor will quietly exit without modifying the document.
+`if` | Optional | A conditional expression to determine whether to execute this processor.
+`on_failure` | Optional	| Specifies how to handle failures for this processor. See the documentation on [Handling failures in pipelines]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
+`ignore_failure` | Optional | If `true`, failures for this processor are ignored. See the documentation on [Handling failures in pipelines]({{site.url}}{{site.baseurl}}/ingest-pipelines/pipeline-failures/).
+`tag` | Optional | An identifier for this processor. Useful for debugging and metrics.
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline named `test-foreach` that uses the `foreach` processor to extract information from a nested JSON structure: 
+
+```json
+PUT _ingest/pipeline/test-foreach
```

Comment:
Using json processor wrapped in a foreach processor is not a valid case, here's an example which uses the foreach processor to lowercase all the elements in an array.

```
PUT _ingest/pipeline/test-foreach
{
  "description": "Lowercase all the elements in an array",
  "processors": [
    {
      "foreach": {
        "field": "protocols",
        "processor": {
          "lowercase": {
            "field": "_ingest._value"
          }
        }
      }
    }
  ]
}
```